src/components/planner/sidebar/CoursesController/ClassItem.tsx

```diff
@@ -49,12 +49,10 @@ const ClassItem = ({ course_id, classInfo, displayed, checked, preview, onSelect
       for (const slot1 of pickedClass.slots)
         for (const slot2 of classInfo.slots)
           if (schedulesConflict(slot1, slot2)) {
-            if (slot1.lesson_type == "TP" && slot2.lesson_type == "TP")
+            if (conflictsSeverity(slot1, slot2))
               return 2
-            else if (slot1.lesson_type == "TP" || slot2.lesson_type == "TP")
-              return 1
             else
-              return 0
+              return 1
```

Comment:
I liked the approach of a `isMandatory` function.

My only suggestion is to return directly the severity degree in `conflictsSeverity`.

Ex:
```
const conflictsSeverity = (first: SlotInfo, second: SlotInfo) => {
  if(isMandatory(first) && isMandatory(second)) return 2
  return 1
}
```